tracer/src/Datadog.Trace/Util/IdGenerator.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="IdGenerator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Util;
+
+/// <summary>
+/// Generates random numbers suitable for use as Datadog trace ids and span ids.
+/// </summary>
+internal class IdGenerator
+{
+    /// <summary>
+    /// Returns a random number greater than zero and less than Int64.MaxValue.
+    /// </summary>
+    public static ulong NextUInt64()
```

Comment:
Just wondering if it would it make more sense to put this into the `ThreadSafeRandom` class?
But that might muddy it up a bit since you've also just cleaned it up :)

Reply:
Yeah, I went back and forth on this so I welcome suggestions. I'm not a fan of having all these related but separate RNG types:
- `ThreadSafeRandom`: a general-purpose safety wrapper around `System.Random`, used in many places
- `RandomNumberGenerator`:  contains the `Xoshiro256StarStar` implementation from `System.Diagnostics.DiagnosticSource`, used only in `IdGenerator`
- `IdGenerator`: generates ids that meet specific rules for trace ids and span ids. Used to rely on `ThreadSafeRandom`, now relies on `System.Random` (>= .NET 6) or `RandomNumberGenerator` (< .NET 6)

Reply:
I made some changes to make things simpler (I hope). I merged the `RandomNumberGenerator` code into `IdGenerator` and renamed it `RandomIdGenerator`. So now there are only 2 types:

- `ThreadSafeRandom`: a general-purpose safety wrapper around `System.Random`
- `RandomIdGenerator`: generates ids that meet specific rules for trace ids and span ids, on all runtimes it uses its own `xoshiro256**` implementation instead of `System.Random`